internal/app/oauth/storage.go

```diff
@@ -0,0 +1,337 @@
+			accessTokenType: op.AccessTokenTypeJWT,
+			responseTypes:   []oidc.ResponseType{oidc.ResponseTypeCode},
+			grantTypes:      []oidc.GrantType{oidc.GrantTypeCode, oidc.GrantTypeRefreshToken},
+			redirectURIs:    []string{"http://localhost:3000"},
+			allowedScopes:   []string{"openid", "profile", "email"},
+			loginURI:        "http://localhost:3000/login?id=%s",
+			iDTokenLifetime: 5 * time.Minute,
+			clockSkew:       0,
+			devMode:         true,
+		},
+	}
+
+	requests = make(map[string]AuthRequest)
+}
+
+func initKeys(s *AuthStorage) {
+
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic(err)
+	}
+
+	s.key = key
+
+	cert := &x509.Certificate{
+		SerialNumber: big.NewInt(1658),
+		Subject: pkix.Name{
+			Organization:  []string{"Company, INC."},
+			Country:       []string{"US"},
+			Province:      []string{""},
+			Locality:      []string{"San Francisco"},
+			StreetAddress: []string{"Golden Gate Bridge"},
+			PostalCode:    []string{"94016"},
+		},
+		IPAddresses:  []net.IP{net.IPv4(127, 0, 0, 1), net.IPv6loopback},
+		NotBefore:    time.Now(),
+		NotAfter:     time.Now().AddDate(10, 0, 0),
+		IsCA:         true,
+		SubjectKeyId: []byte{1, 2, 3, 4, 6},
+		ExtKeyUsage:  []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:     x509.KeyUsageCertSign,
+	}
+
+	caBytes, err := x509.CreateCertificate(rand.Reader, cert, cert, &key.PublicKey, key)
+	if err != nil {
+		panic("failed to create the cert")
+	}
+
+	cert, err = x509.ParseCertificate(caBytes)
+	if err != nil {
+		panic("failed to create the cert")
+	}
+
+	keySet = jose.JSONWebKeySet{
+		Keys: []jose.JSONWebKey{
+			{Key: key.Public(), Use: "sig", Algorithm: "RS256", KeyID: "1", Certificates: []*x509.Certificate{cert}},
+		},
+	}
+}
+
+func (s *AuthStorage) Health(_ context.Context) error {
+	return nil
+}
+
+func (s *AuthStorage) CreateAuthRequest(_ context.Context, authReq *oidc.AuthRequest, _ string) (op.AuthRequest, error) {
+
+	ti := time.Now()
+	entropy := ulid.Monotonic(mRand.New(mRand.NewSource(ti.UnixNano())), 0)
+
+	request := &AuthRequest{
+		ID:           ulid.MustNew(ulid.Timestamp(ti), entropy).String(),
+		ClientID:     authReq.ClientID,
+		subject:      "",
+		code:         "", // Will be set after /authorize/callback success
+		state:        authReq.State,
+		scopes:       authReq.Scopes,
+		ResponseType: authReq.ResponseType,
+		Nonce:        authReq.Nonce,
+		RedirectURI:  authReq.RedirectURI,
+		createdAt:    time.Now().UTC(),
+		authorizedAt: nil,
+	}
+	if authReq.CodeChallenge != "" {
+		request.CodeChallenge = &oidc.CodeChallenge{
+			Challenge: authReq.CodeChallenge,
+			Method:    authReq.CodeChallengeMethod,
+		}
+	}
+
+	requests[request.ID] = *request
+	return request, nil
+}
+
+func (s *AuthStorage) AuthRequestByID(_ context.Context, requestID string) (op.AuthRequest, error) {
+
+	if requestID == "" {
+		return nil, errors.New("invalid id")
+	}
+	request, exists := requests[requestID]
```

Comment:
Ensure this code is called by goroutine-safe func. If not, `sync.Mutex` should be used because modifying maps are not goroutine-safe.